// next-serverless/lib/server.js
"use strict";

var _http = require("http");

var _next = _interopRequireDefault(require("next"));

var _minimist = _interopRequireDefault(require("minimist"));

var _handler = _interopRequireDefault(require("./handler"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const argv = (0, _minimist.default)(process.argv.slice(2), {
  alias: {
    h: 'help',
    H: 'hostname',
    p: 'port'
  },
  boolean: ['h'],
  string: ['H'],
  default: {
    p: process.env.PORT || 3000
  }
});
const dev = process.env.NODE_ENV !== 'production';
const dir = argv._[argv._[0] === 'dev' || argv._[0] === 'start' ? 1 : 0] || '.';
const app = (0, _next.default)({
  dev,
  dir
});
const handler = app.getRequestHandler();
module.exports.handler = (0, _handler.default)(app, handler, () => {
  // create a normal http node server for local usage
  (0, _http.createServer)(handler).listen(argv.port, argv.hostname, err => {
    if (err) throw err;
    console.log(`> Ready on http://${argv.hostname || 'localhost'}:${argv.port}`);
  });
});
